### PR TITLE
Add core concepts and re-arrange nav

### PR DESCRIPTION
**Pull request by mckayward**
Adds core concepts page (under Get Started), and rearranges nav. Also added a couple pages and split some others. Being served [here](https://mckayward.github.io/floyd-docs/)

There are a lot of changes here, and I want to get the conversation going and get this merged in before it gets too big.

### CODE REVIEW

File: docs/getstarted/core_concepts.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
+# Core Concepts
+
+FloydHub is a platform that enables data scientists use to create deep learning
+models more efficiently. FloyHub removes the of the hassle of maintaining and
```
Comment (houqp) [inline docs/getstarted/core_concepts.md]: typo: the of the

File: docs/getstarted/core_concepts.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+provisioning deep-learning servers, allowing data scientists to focus on the
+science of deep learning. It's deep learning without the DevOps.
```
Comment (houqp) [inline docs/getstarted/core_concepts.md]: The Home section is already dedicated to platform introduction. Do we still need to add the introduction in this section?

File: docs/getstarted/core_concepts.md (hunk 1 above)
Comment (mckayward) [reply docs/getstarted/core_concepts.md]: fixed

File: docs/getstarted/core_concepts.md (hunk 2 above)
Comment (mckayward) [reply docs/getstarted/core_concepts.md]: Yeah you make a good point. Let me think of a better intro here.

**Pull request merged**